### DISCUSSION TITLE
Facing issue while trigering curl command for any service ,while running services with security using manual commands

### DISCUSSION THREAD

**Discussion by divyashree123456** [edgexfoundry/community#250] [General]
when all the services are up and running in security mode using manual commands
Im facing "level=WARN ts=2024-02-08T18:12:59.78741Z app=core-metadata source=auth_middleware.go:67 msg="Request to '/api/v3/version' UNAUTHORIZED" in **core metadata** when i m triggering curl command 
`root@NB2-SOC-BSP-14DEC23-1:~/edgex-go-napa/cmd/security-bootstrapper# curl  "http://localhost:59881/api/v3/version"
{"message":"Unauthorized"}
`
I have seen document related to authentication on edgex foundry ,but still not getting where can i get JWT token which is used to parse in  curl command.
please help me to resolve this issue.
thanks in advance.

**Comment by lenny-goodell** [comment]
See the following doc links
https://docs.edgexfoundry.org/3.1/security/Ch-Authenticating/#3-obtaining-a-jwt-authentication-token
https://docs.edgexfoundry.org/3.1/security/Ch-Authenticating/#calls-to-local-services

You can also use edgex-compose `make get-token` command to generate the JWT. See READMEs here:
https://github.com/edgexfoundry/edgex-compose/blob/main/README.md#compose-files
https://github.com/edgexfoundry/edgex-compose/blob/main/compose-builder/README.md#get-token

**Comment by sanketpatil5435** [comment]
@divyashree123456   Have you solved this issue? I am also facing the same.

**Reply by cloudxxx8** [reply]
The document from Lenny is quite clear.  You just need JWT if the security is enabled.

**Comment by divyashree123456** [comment]
Hi @sanketpatil5435  ya it got resolved by creating a JWT authentication tokens for services .

**Reply by sanketpatil5435** [reply]
@divyashree123456 
I'm able to generate the token but when I call the error pops up as
Error- Object { headers: {…}, status: 503, statusText: "Service Unavailable", url: "http://localhost:4000/core-metadata/api/v3/ping", ok: false, name: "HttpErrorResponse", message: "Http failure response for http://localhost:4000/core-metadata/api/v3/ping: 503 Service Unavailable", error: "HTTP request timeout" }
​
error: "HTTP request timeout"

make get-token generated token is - eyJhbGciOiJFUzM4NCIsImtpZCI6ImUyNzJjY2U5LTc4M2ItZjg1Yy1iZDljLTdiMGE1YzJkMDc2OCJ9.eyJhdWQiOiJlZGdleCIsImV4cCI6MTcyNjU1NTA5NywiaWF0IjoxNzI2NTUxNDk3LCJpc3MiOiIvdjEvaWRlbnRpdHkvb2lkYyIsIm5hbWUiOiJlZGdleHVzZXIiLCJuYW1lc3BhY2UiOiJyb290Iiwic3ViIjoiOTZhNzYzNDctMWU2Mi0yNzY2LTdjYjItMmMxNDBiZDEwOGNmIn0.O0-dg-Ai4ltHckFtQtaHqPpoDluIzWnJKty42SNr43pGI-vcQOORoT8vQFZ1QYhedb0MQU0bBsch-v7GAY4bHWv0YbXdbTYhBbUdX9m7ICIvg0KqydCCDB2zsYtI2Fst

**Reply by cloudxxx8** [reply]
The port of API Gateway should be 8443, not 4000, and you should use https instead of http
https://docs.edgexfoundry.org/3.1/security/Ch-APIGateway/
